File: php.local/www/experiment/compilation/literal/viaParam.php
<?php
// viaParam.php

$operator = $_GET['operator'] ?? 'divide';
$divisor = $_GET['divisor'] ?? 0;
try {

	if ($operator == 'divide'){
		$result = 1 / $divisor;
	}else{
		$result = 1 % $divisor;
	}
} catch (Throwable $t){
	echo "Operator: $operator<br>";
	printf("Type: %s<br>", get_class($t));
	printf("Message: %s<br>", $t->getMessage());
}

File: php.local/www/experiment/compilation/literal/viaParam.rb
## viaParam.rb

operator = ARGV[0] || "divide"
divisor = (ARGV[1] || 1).to_i
begin
	if operator == 'divide' then
		result = 1 / divisor
	else
		result = 1 % divisor
	end
rescue Exception => e
	puts "Operator: #{operator}"
	puts "Message: #{e.exception}"
	puts "Class: #{e.class}"
end
